### PR TITLE
[SPARK-13356][Streaming]WebUI missing input informations when recovering from dirver failure

### PR DESCRIPTION
**Pull request by jeanlyn**
WebUI missing some input information when streaming recover from checkpoint, it may confuse people the data lost when restarting streaming context.For example:
directkafkascreenshot

To address this issue, i try to:
-  Invoke `reportInfo` when restarting streaming context before generating rdd, 
-  Invoke the `generateJobs` using the info of `inputInfoTracker` to submit the job when restarting the `JobGenerator`. 

more detail:[SPARK-13356](https://issues.apache.org/jira/browse/SPARK-13356)

### CODE REVIEW

**Comment by jeanlyn** [comment]
@tdas @zsxwing

**Comment by JoshRosen** [comment]
Can you add a descriptive description to this pull request?

**Comment by jeanlyn** [comment]
@JoshRosen Sure.

**Comment by zsxwing** [comment]
ok to test

**Comment by andrewor14** [comment]
@zsxwing @tdas

**Comment by vanzin** [comment]
@jeanlyn do you mind updating the PR to solve conflicts? thanks

**Comment by jeanlyn** [comment]
@vanzin Sorry for the late reply, I had solved the conflicts.

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala [hunk 1]
```diff
@@ -181,6 +180,11 @@ class StreamingContext private[streaming] (
   }
 
   private[streaming] val scheduler = new JobScheduler(this)
+  // restore data has to report info to inputInfoTracker,
+  // so it need to be initialzed after JobScheduler
+  if(isCheckpointPresent) {
```
Comment (vanzin) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: nit: `if (`

**Comment by vanzin** [comment]
@tdas @koeninger could one of you guys take a look? This is not really my area.

**Comment by tdas** [comment]
Hello @jeanlyn, I apologize for not seeing this PR for so long. This is a very good bug to fix, thanks a lot. However, this needs additional unit tests to avoid this bug in the future. So if are still available to work on this PR, could you add unit tests?

**Comment by jeanlyn** [comment]
@tdas OK, i will try to add unit test these day.

**Comment by jeanlyn** [comment]
@tdas I had added a unit test to this PR, could you take some time to review ?

**Pull request closed without merging**